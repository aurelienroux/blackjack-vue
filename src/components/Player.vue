<template>
  <div class="player">
    <p>Player: {{ playerScore }}</p>
    <div class="display">
      <div v-for="(card, index) in playerCards" :key="index">
        <img :src="card.image" alt="player card" />
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Player',
  props: ['playerScore', 'playerCards']
}
</script>

<style lang="scss">
.player {
  margin-bottom: 1rem;
  padding: 0 2rem;

  p {
    margin: 0.5rem 0 0.25rem;
  }
}

.display {
  display: flex;
  flex-wrap: wrap;
  min-height: 6.5rem;

  @include for-tablet-portrait-up {
    min-height: 8.5rem;
  }
}
</style>
