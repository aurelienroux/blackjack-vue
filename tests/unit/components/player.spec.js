import { shallowMount } from '@vue/test-utils'
import Player from '@/components/Player.vue'

function createLocalWrapper(playerScore = 0, playerCards = []) {
  return shallowMount(Player, {
    propsData: {
      playerScore,
      playerCards
    }
  })
}

const playerCards = [
  {
    code: '5C',
    image: 'https://deckofcardsapi.com/static/img/5C.png',
    images: {
      svg: 'https://deckofcardsapi.com/static/img/5C.svg',
      png: 'https://deckofcardsapi.com/static/img/5C.png'
    },
    value: '5',
    suit: 'CLUBS'
  },
  {
    code: '3S',
    image: 'https://deckofcardsapi.com/static/img/3S.png',
    images: {
      svg: 'https://deckofcardsapi.com/static/img/3S.svg',
      png: 'https://deckofcardsapi.com/static/img/3S.png'
    },
    value: '3',
    suit: 'SPADES'
  }
]

describe('Player component', () => {
  describe('default state', () => {
    it('renders', () => {
      const wrapper = createLocalWrapper()
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('p').text()).toMatch('Player: 0')
    })
  })

  describe('playing', () => {
    it('renders', () => {
      const wrapper = createLocalWrapper(10, playerCards)
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('p').text()).toMatch('Player: 10')
      expect(wrapper.findAll('img').length).toBe(2)
      expect(
        wrapper
          .findAll('img')
          .at(0)
          .attributes('src')
      ).toBe('https://deckofcardsapi.com/static/img/5C.png')
      expect(
        wrapper
          .findAll('img')
          .at(1)
          .attributes('src')
      ).toBe('https://deckofcardsapi.com/static/img/3S.png')
    })
  })
})
