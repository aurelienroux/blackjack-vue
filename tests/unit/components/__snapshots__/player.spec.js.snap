// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component default state renders 1`] = `
<div class="player">
  <p>Player: 0</p>
  <div class="display"></div>
</div>
`;

exports[`Player component playing renders 1`] = `
<div class="player">
  <p>Player: 10</p>
  <div class="display">
    <div><img src="https://deckofcardsapi.com/static/img/5C.png" alt="player card"></div>
    <div><img src="https://deckofcardsapi.com/static/img/3S.png" alt="player card"></div>
  </div>
</div>
`;
